### PR TITLE
MGDAPI-5081 - RHOAM changes for GCP

### PR DESCRIPTION
**Pull request by adam-cattermole**
# Issue link
[MGDAPI-5081](https://issues.redhat.com/browse/MGDAPI-5081)

# What
- Update to cloud resources product to reconcile correct strategy
- RHSSO expects a DNS name for Postgres instances rather than IP so experimental configuration enables connection
- Threescale refactor to avoid AWS specific implementation on GCP
- Built a temporary index image for 3scale with system component changes, alongside a CRO image from gcp dev branch for testing
- Removed the croaws dependency from `make cluster/prepare/local` as the strategy is created by RHOAM anyway
- Reverted the change to use `kube_endpoint_address` instead of `kube_endpoint_address_available` in MCG alerts. We have seen on an E2E run that `kube_endpoint_address` was not present on the cluster and was always firing.

**CRO PR integr8ly/cloud-resource-operator#634 should be merged first, and the replaces updated to refer to the main integreatly gcp branch.**

# Verification steps
### Provision a GCP CCS cluster
If you do not have an access key,  request one from myself - place in root of delorean repo and run:
```
OCM_CLUSTER_NAME=acatterm-ccs OCM_CLUSTER_LIFESPAN=24 COMPUTE_NODES_COUNT=4 BYOC=true CLOUD_PROVIDER=gcp make -f make/ocm.mk ocm/cluster.json
make -f make/ocm.mk ocm/cluster/create
```

### Deploy RHOAM from this branch

useClusterStorage should be set to false.

```
LOCAL=false make cluster/prepare/local
LOCAL=false USE_CLUSTER_STORAGE=false make code/run
```

### Verify installation

Watch RHMI CR and operator logs and check that installation completes successfully. Check in the GCP cloud account that the Postgres/Redis resources are present

### Verify uninstallation

Watch RHMI CR and oeprator logs and check uninstallation completes successfully. We expect the resources in the GCP cloud account to be removed/cleaned up during this process.

### CODE REVIEW

**Review by cecobask — APPROVED** [review]
Watched the cloud resources being provisioned and cleaned up by cro.  
RHOAM installed and uninstalled successfully and there was nothing out of the ordinary in the logs.  
Completed all verification steps successfully.  
The code changes look good!  

/lgtm

File: pkg/products/rhssocommon/reconciler.go [hunk 1]
```diff
@@ -832,3 +834,36 @@ func (r *Reconciler) ReconcilePodDisruptionBudget(ctx context.Context, apiClient
...
+					Name: resources.RHSSOPostgresServiceHost,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: pgresSelector,
+					},
+				},
+			},
+		}, nil
+	}
+	return keycloak.ExperimentalSpec{}, nil
+}
```
[the remark this answers is not in the document]
Comment (adam-cattermole) [reply pkg/products/rhssocommon/reconciler.go]: Updated in latest commit 81e577f

**Comment by adam-cattermole** [comment]
@cecobask I've added one commit where I moved the cro module from my branch to the main GCP branch - if you're happy could you re-lgtm

**Pull request merged**